[quick_actions] add localizedSubtitle for iOS

sinyu1012 opened the pull request:
Add the localizedSubtitle field on quick actions for iOS

Fixes https://github.com/flutter/flutter/issues/129759

Pre-launch Checklist
- [x] I read the [Contributor Guide](https://github.com/flutter/packages/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md) page, which explains my responsibilities.
 I read and followed the [relevant style guides](https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style) and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use dart format.)
- [x] I signed the [CLA](https://cla.developers.google.com/).
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. [shared_preferences]
- [x] I [linked to at least one issue that this PR fixes](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview) in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning), or this PR is [exempt from version changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version).
- [x] I updated CHANGELOG.md to add a description of the change, [following repository CHANGELOG style](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style), or this PR is [exempt from CHANGELOG changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog).
- [x] I updated/added relevant documentation (doc comments with ///).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests).
- [x] All existing and new tests are passing.


If you need help, consider asking for advice on the #hackers-new channel on [Discord](https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md).

jmagman commented on packages/quick_actions/quick_actions_ios/example/pubspec.yaml:
```diff
@@ -25,3 +25,8 @@ dev_dependencies:
 
```
Can you also update the example app to use `localizedSubtitle` like you did in `packages/quick_actions/quick_actions/example/lib/main.dart`?

Around here:
https://github.com/flutter/packages/blob/67d8b50c1796eefa948a1367c1184b0e71753e44/packages/quick_actions/quick_actions_ios/example/lib/main.dart#L53-L54

```dart
        localizedTitle: 'Action one',
        localizedSubtitle: 'Action one subtitle',
        icon: 'AppIcon',
```

Then the native test here
https://github.com/flutter/packages/blob/67d8b50c1796eefa948a1367c1184b0e71753e44/packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift#L77-L78

becomes:
```swift
    findAndTapQuickActionButton(
      buttonName: "Action one, Action one subtitle", quickActionsAppIcon: quickActionsAppIcon, springboard: springboard)

```
In the XCUITest you can `p springboard` to see the label is joining the title and subtitle with a comma:
```
                      Button, 0x107906630, {{31.0, 260.7}, {250.0, 63.7}}, identifier: 'action_one', label: 'Action one, Action one subtitle'
```

jmagman commented:
Also, `Mac_arm64 macos_repo_checks ` is showing a swift-format error:
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8731849849150109729/+/u/Run_package_tests/Swift_format/stdout

sinyu1012 commented:
> Also, `Mac_arm64 macos_repo_checks ` is showing a swift-format error: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8731849849150109729/+/u/Run_package_tests/Swift_format/stdout

Thanks，Please help review again.

jmagman reviewed (approved):
quick_actions_ios part LGTM.

@stuartmorgan should I just remove the `triage-ios` label to kick it back to be reviewed for the federated interface part?

stuartmorgan commented:
> @stuartmorgan should I just remove the `triage-ios` label to kick it back to be reviewed for the federated interface part?

We don't filter out by `triage-*` labels, so the other reviews should happen in parallel; you can just remove it once the iOS team doesn't expect to need to provide any further input on a PR.

sinyu1012 commented:
@stuartmorgan  What do I need to do next?

bparrishMines commented on packages/quick_actions/quick_actions_android/example/pubspec.yaml:
```diff
@@ -26,3 +26,4 @@ dev_dependencies:
 
 flutter:
   uses-material-design: true
+
```
```suggestion
```

bparrishMines commented on packages/quick_actions/quick_actions_android/pubspec.yaml:
```diff
@@ -33,3 +33,4 @@ dev_dependencies:
 topics:
   - quick-actions
   - os-integration
+
```
```suggestion
```

bparrishMines commented on packages/quick_actions/quick_actions/example/lib/main.dart:
```diff
@@ -53,14 +53,16 @@ class _MyHomePageState extends State<MyHomePage> {
         localizedTitle: 'Action one',
-        icon: 'AppIcon',
+        localizedSubtitle: 'Action one subtitle',
+        icon: 'icon_help',
```
Why was the icon changed here?

sinyu1012 replied:
I changed it back to the previous `AppIcon`.

bparrishMines reviewed (approved):
LGTM

This generally looks good to me with just a few nits.

You can move to the next step and create the `quick_action_platform_interface` PR.

sinyu1012 commented:
> LGTM
> 
> This generally looks good to me with just a few nits.
> 
> You can move to the next step and create the `quick_action_platform_interface` PR.

@bparrishMines  Thank you, I have made corresponding modifications to address the points you raised. What should I do next.

stuartmorgan commented:
> What should I do next.

See https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changing-federated-plugins for the steps involved in landing a multi-package PR.

sinyu1012 commented:
> > What should I do next.
> 
> See https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changing-federated-plugins for the steps involved in landing a multi-package PR.

@stuartmorgan  OK, thank you! I have submitted a new PR: https://github.com/flutter/packages/pull/8112, please help review

sinyu1012 commented:
@stuartmorgan @bparrishMines  @jmagman  @camsim99 

The prerequisite PR (flutter/packages#8112) for the platform interface has been merged. I have:
1. Updated the dependencies to use the published version (^1.1.0) of quick_actions_platform_interface
2. Removed all path-based dependency overrides
3. Verified that all tests are passing

Could you please review this PR again? Thank you for your time!

stuartmorgan commented on packages/quick_actions/quick_actions/pubspec.yaml:
```diff
@@ -22,7 +22,7 @@ dependencies:
     sdk: flutter
   quick_actions_android: ^1.0.0
   quick_actions_ios: ^1.0.0
```
The iOS part needs to land in a separate PR and this constraint updated to require it, otherwise someone who updates by editing their pubspec.yaml instead of running `pub update` will get a set of versions that claims to support `localizedTitle`, allows them to pass `localizedTitle`, but ignores that parameter. See [step 5 of the instructions for federated plugins](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changing-federated-plugins).

sinyu1012 replied:
Thank you for the feedback. I have created this [PR]（ https://github.com/flutter/packages/pull/8149 ).

stuartmorgan commented on packages/quick_actions/quick_actions/CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## 1.0.9
```
A new feature is a minor version change.

stuartmorgan commented on packages/quick_actions/quick_actions_ios/CHANGELOG.md:
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 1.1.2
```
A new feature is a minor version change.

sinyu1012 commented:
@stuartmorgan @bparrishMines @jmagman @camsim99

The prerequisite PR (https://github.com/flutter/packages/pull/8149) for the ios package has been merged. Could you please review this PR again? Thank you for your time!

stuartmorgan reviewed (approved):
LGTM
